### PR TITLE
[SIMS#562][Forms]NOA formio changes for PartTime

### PR DESCRIPTION
**Pull request by guru-aot**
NOA form changes

PartTime:
image
image
image

FullTime:
image
image
image
image

### CODE REVIEW

File: sources/packages/forms/src/form-definitions/noticeofassessment.json [hunk 1]
```diff
@@ -2397,13 +2546,21 @@
...
+          "label": "HTML",
+          "tag": "p",
+          "className": "",
+          "attrs": [
+            {
+              "attr": "",
+              "value": ""
+            }
+          ],
+          "content": "<p style=\\\"margin-left:0cm;\\\">Your application was subject to the federal and provincial financial need assessment calculations, based on Canada Student Loan Program and StudentAid BC’s respective policies</p><ul><li>Total family income on the previous year's returns: <b>${{ data.assessment?.totalFamilyIncome }}</b></li><li>This Includes the income of your spouse, parents, stepparents, sponsors, or legal guardians</li><li>Scholarships and bursaries reported</li><li>Voluntary or targeted contributions reported, and</li><li>Assets reported by parents, stepparents, sponsors, or legal guardians (dependent students only)</li></ul>",
```
Comment (ann-aot) [inline sources/packages/forms/src/form-definitions/noticeofassessment.json]: for margin, we can use bootstrap classes(https://getbootstrap.com/docs/4.0/utilities/spacing) instead of inline classes, if the class is not present you can create custom class

File: sources/packages/forms/src/form-definitions/noticeofassessment.json [hunk 2]
```diff
@@ -1755,7 +1904,7 @@
                         "value": ""
                       }
                     ],
-                    "content": "Date 1: THE DATE OF DISBURSEMENT 1",
+                    "content": "Date 1: The ealiest date you will receive funding",
```
Comment (andrewsignori-aot) [inline sources/packages/forms/src/form-definitions/noticeofassessment.json]: I believe that there is a typo on "earliest".

File: sources/packages/forms/src/form-definitions/noticeofassessment.json [hunk 3]
```diff
@@ -1923,7 +2072,7 @@
                         "value": ""
                       }
                     ],
-                    "content": "Date 2 (If applicable): THE DATE OF DISBURSEMENT 2",
+                    "content": "Date 2 (If applicable): The ealiest date you will receive funding",
```
Comment (andrewsignori-aot) [inline sources/packages/forms/src/form-definitions/noticeofassessment.json]: I believe that there is a typo on "earliest".

File: sources/packages/forms/src/form-definitions/noticeofassessment.json [hunk 4]
```diff
@@ -3078,43 +3228,24 @@
                     ],
                     "content": "{{typeof data.assessment?.tuitionCost === 'number' ? '$' + (data.assessment.tuitionCost).toFixed(2) : '-'}}",
```
Comment (ann-aot) [inline sources/packages/forms/src/form-definitions/noticeofassessment.json]: this is not a part of this PR @guru-aot , I think `.toFixed()` was having some issue, it was not rounding correctly, in some scenarios. remember @dheepak-aot @andrewsignori-aot ?

File: sources/packages/forms/src/form-definitions/noticeofassessment.json [hunk 5]
```diff
@@ -3732,43 +3852,24 @@
                     ],
                     "content": "{{typeof data.assessment?.livingAllowance === 'number' ? '$' + (data.assessment.livingAllowance).toFixed(2) : '-'}}",
```
Comment (ann-aot) [inline sources/packages/forms/src/form-definitions/noticeofassessment.json]: not part of this PR, we do have javascript builts functions to check number like `Number.isInteger`

File: sources/packages/forms/src/form-definitions/noticeofassessment.json [hunk 6]
```diff
@@ -6963,25 +6896,23 @@
...
                     "attrs": [
                       {
                         "attr": "",
                         "value": ""
                       }
                     ],
-                    "content": "Canada Student Loan",
-                    "refreshOnChange": false,
-                    "key": "canadaStudentLoan",
+                    "content": "{{typeof data.assessment?.booksAndSuppliesCost === 'number' ? '$' + (data.assessment.booksAndSuppliesCost).toFixed(2) : '-'}}",
```
Comment (ann-aot) [inline sources/packages/forms/src/form-definitions/noticeofassessment.json]: added comments above

File: sources/packages/forms/src/form-definitions/noticeofassessment.json [hunk 7]
```diff
@@ -7700,50 +7605,30 @@
...
-                    "attributes": {},
-                    "overlay": {
-                      "style": "",
-                      "page": "",
-                      "left": "",
-                      "top": "",
-                      "width": "",
-                      "height": ""
-                    },
+                    "content": "<b>Total assessed costs</b> = Total Academic Expenses + Childcare Costs",
```
Comment (ann-aot) [inline sources/packages/forms/src/form-definitions/noticeofassessment.json]: i think for bold, you can create a custom class as per the figma, if not there

File: sources/packages/forms/src/form-definitions/noticeofassessment.json (hunk 2 above)
Comment (dheepak-aot) [inline sources/packages/forms/src/form-definitions/noticeofassessment.json]: earliest?

File: sources/packages/forms/src/form-definitions/noticeofassessment.json (hunk 3 above)
Comment (dheepak-aot) [inline sources/packages/forms/src/form-definitions/noticeofassessment.json]: same typo here.

File: sources/packages/forms/src/form-definitions/noticeofassessment.json (hunk 5 above)
Comment (guru-aot) [reply sources/packages/forms/src/form-definitions/noticeofassessment.json]: The reason why we have typeOf and not Number.isInteger is Number.isInteger(2.03) return false and typeof 3.14                   // Returns "number". In the assessment json we will have values with decimals.

File: sources/packages/forms/src/form-definitions/noticeofassessment.json (hunk 6 above)
Comment (guru-aot) [reply sources/packages/forms/src/form-definitions/noticeofassessment.json]: Please check comments above

File: sources/packages/forms/src/form-definitions/noticeofassessment.json (hunk 4 above)
Comment (dheepak-aot) [reply sources/packages/forms/src/form-definitions/noticeofassessment.json]: Here toFixed() has been used to display as 2 decimal positions(e.g 20 as 20.00) these values are already rounded with Math.Round as per our discussion in camunda.

File: sources/packages/forms/src/form-definitions/noticeofassessment.json (hunk 5 above)
Comment (dheepak-aot) [reply sources/packages/forms/src/form-definitions/noticeofassessment.json]: The purpose of using this logic is, to differentiate between 0 and undefined. Yes Number.isInteger can be used only if we get integers from the service. We might get non integers as well like 115.50. We even tried Number.isNan it was returning false for undefined surprisingly.

**Review by ann-aot — COMMENTED** [review]
added some comments @guru-aot  :) LGTM 👍

File: sources/packages/forms/src/form-definitions/noticeofassessment.json (hunk 2 above)
Comment (dheepak-aot) [reply sources/packages/forms/src/form-definitions/noticeofassessment.json]: Oops. I didn't notice @andrewsignori-aot mentioned the same comment. Nevermind.

**Comment by dheepak-aot** [comment]
LGTM. Please look at the review comments.

File: sources/packages/forms/src/form-definitions/noticeofassessment.json (hunk 4 above)
Comment (andrewsignori-aot) [reply sources/packages/forms/src/form-definitions/noticeofassessment.json]: I checked also on Camunda @ann-aot, the rounding there is happening as expected.
So, what @dheepak-aot is saying is when we have a value like 123 the to fixed will make it 123.00 or 123.9 will be 123.90.

File: sources/packages/forms/src/form-definitions/noticeofassessment.json (hunk 5 above)
Comment (ann-aot) [reply sources/packages/forms/src/form-definitions/noticeofassessment.json]: mm, okay, i got it @dheepak-aot

Comment (andrewsignori-aot) [reply sources/packages/forms/src/form-definitions/noticeofassessment.json]: @ann-aot I talked with @guru-aot and we tested with isNaN. It seem to be better. He is going to change some code and share to see how it looks like.

File: sources/packages/forms/src/form-definitions/noticeofassessment.json [hunk 8]
```diff
@@ -7471,12 +7382,12 @@
                         "value": ""
                       }
                     ],
-                    "content": "{{data.disbursement?.disbursement2bcsl ? '$'+ data.disbursement.disbursement2bcsl : '-'}}",
+                    "content": "${{((typeof data.assessment?.tuitionCost === 'number' ? data.assessment?.tuitionCost : 0) \n        + (typeof data.assessment?.booksAndSuppliesCost === 'number' ? data.assessment?.booksAndSuppliesCost : 0)\n        + (typeof data.assessment?.transportationCost === 'number' ? data.assessment?.transportationCost : 0)\n        + (typeof data.assessment?.miscellaneousCost === 'number' ? data.assessment?.miscellaneousCost : 0)).toFixed(2)}}",
```
Comment (andrewsignori-aot) [inline sources/packages/forms/src/form-definitions/noticeofassessment.json]: Please use the total assessment as we talked 👍

**Review by andrewsignori-aot — COMMENTED** [review]
Nice work, please take a look at the comments 👍

File: sources/packages/forms/src/form-definitions/noticeofassessment.json (hunk 1 above)
Comment (guru-aot) [reply sources/packages/forms/src/form-definitions/noticeofassessment.json]: Removed the style, as not needed

File: sources/packages/forms/src/form-definitions/noticeofassessment.json (hunk 5 above)
Comment (ann-aot) [reply sources/packages/forms/src/form-definitions/noticeofassessment.json]: mm, actually, i also did some couple testing, here are the results, 
image

Comment (guru-aot) [reply sources/packages/forms/src/form-definitions/noticeofassessment.json]: For an example scenario
{{(isNaN(data.assessment?.transportationCost) ? '-' : '$' + (data.assessment.transportationCost).toFixed(2))}}

when transportationCost is null, then it will throw "null is not an object (evaluating '(data.assessment.transportationCost).toFixed')."

Not changing as typeOf works for most of the scenarios

**Review by henryoforeh-aot — APPROVED** [review]

**Review by dheepak-aot — APPROVED** [review]
LGTM @guru-aot

**Review by ann-aot — APPROVED** [review]
👍

**Review by andrewsignori-aot — APPROVED** [review]
Thanks for doing the changes. Look good 👍

**Review by abschwenker — APPROVED** [review]
